t/unit-tests/u-oidmap.c

```diff
@@ -0,0 +1,152 @@
+#include "unit-test.h"
+#include "lib-oid.h"
+#include "oidmap.h"
+#include "hash.h"
+#include "hex.h"
+
+/*
+ * Elements we will put in oidmap structs are made of a key: the entry.oid
+ * field, which is of type struct object_id, and a value: the name field (could
+ * be a refname for example).
+ */
+struct test_entry {
+	struct oidmap_entry entry;
+	char name[FLEX_ARRAY];
+};
+
+static const char *const key_val[][2] = { { "11", "one" },
+					  { "22", "two" },
+					  { "33", "three" } };
+
+static void setup(void (*f)(struct oidmap *map))
```

Comment:
Instead of using a `setup()` function, let's use a `test_oidmap__initialize()` function that gets executed automatically and initializes a global variable, as well as a `test_oidmap__cleanup()` function that cleans up state. When we have that we can get rid of `setup()` and also of functions like `test_oidmap__replace()`, where `t_replace_()` itself would be renamed.